client/src/redux/snake/snakeActionCreators.js

```diff
@@ -108,10 +108,12 @@ export const writeOwnSnakePosition = id => (
 
 export const checkForCollisions = id => (
   (dispatch) => {
-    const ownSnake = store.getState().snakes[id];
+    const snakes = store.getState().snakes;
```

Comment:
You don't need to get the snakes from the store because the snake objects are stored in the board itself.  Just pass the snake object that you find in the board to the `collisionType` function.